modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -269,9 +269,13 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
 
             restoreGlobalStateFromLocalMetastorage(recoveryRevision);
 
+            // It is safe to get the latest version of the catalog as we are in the starting process.
+            // Safe to get the latest version until lwm mechanism is implemented for zone lifecycle
+            int catalogVersion = catalogManager.latestCatalogVersion();
```

Comment:
Are you sure? I mean that CatalogManagerImpl is an async starting component that actually does self recovery within 
```
    public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
        int objectIdGen = 0;

        Catalog emptyCatalog = new Catalog(0, 0L, objectIdGen, List.of(), List.of(), null);

        registerCatalog(emptyCatalog);

        updateLog.registerUpdateHandler(new OnUpdateHandlerImpl());

        return updateLog.startAsync(componentContext)
                .thenComposeAsync(none -> {
                ...
```
If my guess is correct, feel free to create a separate ticket for the issue.